bgpd/bgp_routemap.c

```diff
@@ -2422,21 +2428,60 @@ route_set_aspath_replace(void *rule, const struct prefix *dummy, void *object)
 		configured_asn = path->peer->change_local_as
 					 ? path->peer->change_local_as
 					 : path->peer->local_as;
+	} else if (!strncmp(replace, asp_acl, strlen(asp_acl))) {
+		/* its as-path-acl-list command get the access list name */
+		while (*buf == ' ')
+			buf++;
+		buf_acl_name = buf;
+		buf = strchr(buf_acl_name, ' ');
+		if (buf)
+			acl_list_name_len = buf - buf_acl_name;
+		else
+			acl_list_name_len = strlen(buf_acl_name);
+
+		memcpy(acl_list_name, buf_acl_name, acl_list_name_len);
```

Comment:
we need a null terminated string for as_list_lookup. if we change the replace string (adding /0") this will have an impact later. I tried to use the previous conception, rather changing all.

Reply:
taken into account